solc/CommandLineParser.cpp

```diff
@@ -1187,7 +1195,7 @@ General Information)").c_str(),
 	if (m_options.input.mode == InputMode::Compiler)
 		m_options.input.errorRecovery = (m_args.count(g_strErrorRecovery) > 0);
 
-	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
+	solAssert(m_options.input.mode == InputMode::AssemblyJson || m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
```

Comment:
Does this new input mode support all of the arguments that the compiler does? If not, I think it should make the function return at a much earlier point.

Also, which other command-line flags are usable in this mode? There's an issue underway to report errors for flags invalid in other modes (#11629) and I think we should start rejecting invalid flags in this new mode from the very beginning.

Each input mode has a test in `test/solc/CommandLineParser.cpp` that checks all the options valid in that mode. This new mode should have such a test too.

Reply:
I added a very simple check in `solc/CommandLineParser.cpp`. Basically with the new import mode only `--asm`, `--bin` and `--opcodes` are right now supported.